src/models/AbstractWidget.php

```diff
@@ -0,0 +1,126 @@
+<?php namespace Leean\Widgets\Models;
+
+use Leean\Acf\All;
+
+/**
+ * Class AbstractWidget.
+ *
+ * @package Leean\Widgets\Models
+ */
+class AbstractWidget extends \WP_Widget
+{
+	/**
+	 * The slug.
+	 *
+	 * @var bool|string
+	 */
+	private $_slug = '';
+
+	/**
+	 * AbstractWidget constructor.
+	 *
+	 * @param string $title       Widget title
+	 * @param string $description Widget description.
+	 * @param bool   $slug		  Widget slug (use class name if not supplied).
+	 */
+	public function __construct( $title, $description, $slug = false ) {
```

Comment:
No, that line is getting an instance of the child class. That's why I do ti, because I want the child class' name.
I don't see why it has to have an abstract method, abstract just means it can't be instantiated, it has to be inherited. But I don't see why you can't have some default code for each function.
What abstract method would you add?